src/inmanta/ast/type.py

```diff
@@ -229,10 +229,7 @@ def __eq__(self, other: object) -> bool:
 @stable_api
 class Number(Primitive):
```

Comment:
You're missing the bugfix for `number()` casts and associated change entry. I understand the "Do not apply any fixes for it" part may have been misleading here. We can decide to keep the bug, but then we must also revert the fix in iso6 to keep behavior consistent. I think I prefer to just apply the fix. See [Slack](https://inmanta.slack.com/archives/CKRF0C8R3/p1700836622599319).

Reply:
Ok i'll apply the fix. I prefer it with the fix to